From acd322f66f0e9329b5fdc8cc8fc1bf2744e7c3b2 Mon Sep 17 00:00:00 2001
From: Alon Rotman <alon.rotman@solid-run.com>
Date: Mon, 22 Nov 2021 13:22:25 +0200
Subject: [PATCH 2/2] UBOOT: cn9130 som support clearfog base and pro

device tree and makefile to support cn9130 som by solidrun on clearog
base and pro platforms
---
 arch/arm/dts/Makefile           |   4 +-
 arch/arm/dts/cn9130-cf-base.dts | 102 ++++++++++++++++++-------------
 arch/arm/dts/cn9130-cf-pro.dts  | 104 +++++++++++++++++++-------------
 arch/arm/dts/cn9130-som.dtsi    |   2 +
 4 files changed, 125 insertions(+), 87 deletions(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 1b46829593..246bd4665e 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -241,8 +241,8 @@ dtb-$(CONFIG_ARCH_MVEBU) +=			\
 	cn9132-db-B.dtb				\
 	cn9132-db-C.dtb				\
 	cn9132-cex7-A.dtb			\
-	cn9130-cf-base.dtb			\
-	cn9130-cf-pro.dtb
+	cn9130-cf-pro.dtb			\
+	cn9130-cf-base.dtb
 
 dtb-$(CONFIG_ARCH_UNIPHIER_LD11) += \
 	uniphier-ld11-global.dtb \
diff --git a/arch/arm/dts/cn9130-cf-base.dts b/arch/arm/dts/cn9130-cf-base.dts
index 1e65712e34..a9d4a4491f 100644
--- a/arch/arm/dts/cn9130-cf-base.dts
+++ b/arch/arm/dts/cn9130-cf-base.dts
@@ -66,12 +66,19 @@
 					3300000 0x0>;
 			};
 		};
+		gpio@440100 {
+			p24 {
+				gpio-hog;
+				gpios = <24 GPIO_ACTIVE_HIGH>;
+				output-high;
+				line-name = "switch_reset";
+			};
+		};
 	};
 };
 
 
 /***** AP related configuration *****/
-
 &ap_pinctl {
 	/* MPP Bus:
 	 * SDIO  [0-10, 12]
@@ -105,39 +112,30 @@
 
 &cp0_pinctl {
 	/* MPP Bus:
-	*	[0-11,]	RGMII1
+	*	[0-1] 	SMI MDC/MDIO
+	*	[2-3]	UART1 TX/RX
+	*	[4-5]	UART1 RTS/CTS
+	*	[6-8]	PTP
+	*	[10-11] GPIO - SPD Strap
 	*	[12-16]	SPI1
-	*	[17-23]  boot straps
-	*	[24]	GPIO - SFP interrupt - eth0
-	*	[25] 	boot strap	
-	*	[26]	GPIO - FAN TACHO
-	*	[27] 	GPIO - DDR EVENT
-	*	[28-30]	PTP
-	*	[31]	GPIO - PWRBTN_N
-	*	[32]	GPIO - CS Status
-	*	[33]	GPIO - BIOS_DIS_OVERRIDE
-	*	[34]	GPIO - SUS_5_N
-	*	[35,36]	I2C1
-	*	[37,38]	I2C0
-	*	[39]	GPIO - PWM
-	*	[40,41]	SMI - MDC/MDIO
-	*	[42,43]	XSMI - MDC/MDIO
-	*	[44-48]	boot strap
-	*	[49]	CP0_VHV_EN
-	*	[50,51]	UART2
-	*	[52]	RCVR_CLK1
-	*	[53]	AP_VHV_EN
-	*	[54]	RCVR_CLK2
-	*	[55-61]	SDIO
-	*	[62]	NC
+	*	[17-26]	GPIO - Boot straps
+	*	[27-34]	GPIO
+	*	[35-36]	I2C1
+	*	[37-38]	I2C1
+	*	[39] 	GPIO
+	*	[40] 	RCVR CLK
+	*	[41]	GPIO VHV_EN
+	*	[43]	SD CARD DT
+	*	[44-55]	RGMII
+	*	[56-61]	SDIO
 	*/
 		   /*   0   1   2   3   4   5   6   7   8   9 */
-	pin-func = <	3   3   3   3   3   3   3   3   3   3
-			3   3   3   3   3   3   3   0   0   0
-			0   0   0   0   0   0   0   0   7   7
-			7   0   0   0   0   2   2   2   2   0
-			0   0   0   0   0   0   0   8   0   8
-			6   6   2   0   2   0xb  0xe 0xe 0xe 0xe
+	pin-func = <	0xa 0xa 8   8   6   6   9   9   9   0
+			0   0   3   3   3   3   3   0   0   0
+			0   0   0   0   0   0   0   0   0   0
+			0   0   0   0   0   2   2   2   2   0
+			2   0   0   1   1   1   1   1   1   1
+			1   1   1   1   1   1  0xe 0xe 0xe 0xe
 			0xe 0xe 0 >;
 };
 
@@ -149,7 +147,7 @@
 	clock-frequency = <100000>;
 	eeprom0: eeprom@50 {
                 compatible = "atmel,24c64";
-                reg = <0x50>;
+                reg = <0x53>;
                 pagesize = <0x20>;
         };
 	/*
@@ -188,11 +186,11 @@
 			input;
 			line-name = "pcie1.0-clkreq";
 		};
-		pcie1_0_w_disable {
+		pcie1_0_perst {
 			gpio-hog;
 			gpios = <3 GPIO_ACTIVE_LOW>;
 			output-low;
-			line-name = "pcie1.0-w-disable";
+			line-name = "pcie1.0-perst";
 		};
 		usb3_ilimit {
 			gpio-hog;
@@ -206,6 +204,20 @@
 			output-high;
 			line-name = "usb3-power";
 		};
+		pcie2_0_w_disable {
+			gpio-hog;
+			gpios = <8 GPIO_ACTIVE_LOW>;
+			output-low;
+			line-name = "pcie2.0-w-disable";
+		};
+
+		pcie2_0_gnss_disable {
+			gpio-hog;
+			gpios = <9 GPIO_ACTIVE_LOW>;
+			output-low;
+			line-name = "pcie2.0-gnss-disable";
+		};
+
 		m2_devslp {
 			gpio-hog;
 			gpios = <11 GPIO_ACTIVE_HIGH>;
@@ -232,7 +244,6 @@
 	no-1-8-v;
 };
 
-
 /* SPI NOR  */
 &cp0_spi1 {
 	pinctrl-names = "default";
@@ -343,6 +354,12 @@
 	status = "okay";
 };
 
+/* 1GE PHY over RGMII */
+&cp0_eth2 {
+	status = "okay";
+//	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+};
 
 &cp0_pinctl {
 	compatible = "marvell,mvebu-pinctrl", "marvell,armada-8k-cpm-pinctrl";
@@ -356,17 +373,18 @@
 		marvell,pins = < 35 36 >;
 		marvell,function = <2>;
 	};
-	cp0_ge1_rgmii_pins: cp0-ge-rgmii-pins-0 {
-		marvell,pins = < 0 1 2 3 4 5 6 7 8 9 10 11>;
-		marvell,function = <3>;
+	cp0_ge2_rgmii_pins: cp0-ge-rgmii-pins-1 {
+                marvell,pins = < 44 45 46 47 48 49 50 51
+                                52 53 54 55 >;
+                marvell,function = <1>;
 	};
 	cp0_sdhci_pins: cp0-sdhi-pins-0 {
 		marvell,pins = < 56 57 58 59 60 61 >;
 		marvell,function = <14>;
 	};
 	cp0_sdhci_cd_pins: cp0-sdhci-cd-pins-0 {
-		marvell,pins = < 55 >;
-		marvell,function = <11>;
+		marvell,pins = < 43 >;
+		marvell,function = <1>;
 	};
 	cp0_spi1_pins: cp0-spi-pins-0 {
 		marvell,pins = < 13 14 15 16 >;
@@ -377,7 +395,7 @@
 		marvell,function = <3>;
 	};
 	cp0_mdio_pins: cp0-mdio-pins {
-                marvell,pins = < 47 49 >;
-                marvell,function = <8>;
+                marvell,pins = < 40 41 >;
+                marvell,function = <10>;
         };
 };
diff --git a/arch/arm/dts/cn9130-cf-pro.dts b/arch/arm/dts/cn9130-cf-pro.dts
index 27d1d3a7ac..fb37dd833d 100644
--- a/arch/arm/dts/cn9130-cf-pro.dts
+++ b/arch/arm/dts/cn9130-cf-pro.dts
@@ -112,39 +112,30 @@
 
 &cp0_pinctl {
 	/* MPP Bus:
-	*	[0-11,]	RGMII1
+	*	[0-1] 	SMI MDC/MDIO
+	*	[2-3]	UART1 TX/RX
+	*	[4-5]	UART1 RTS/CTS
+	*	[6-8]	PTP
+	*	[10-11] GPIO - SPD Strap
 	*	[12-16]	SPI1
-	*	[17-23]  boot straps
-	*	[24]	GPIO - SFP interrupt - eth0
-	*	[25] 	boot strap	
-	*	[26]	GPIO - FAN TACHO
-	*	[27] 	GPIO - DDR EVENT
-	*	[28-30]	PTP
-	*	[31]	GPIO - PWRBTN_N
-	*	[32]	GPIO - CS Status
-	*	[33]	GPIO - BIOS_DIS_OVERRIDE
-	*	[34]	GPIO - SUS_5_N
-	*	[35,36]	I2C1
-	*	[37,38]	I2C0
-	*	[39]	GPIO - PWM
-	*	[40,41]	SMI - MDC/MDIO
-	*	[42,43]	XSMI - MDC/MDIO
-	*	[44-48]	boot strap
-	*	[49]	CP0_VHV_EN
-	*	[50,51]	UART2
-	*	[52]	RCVR_CLK1
-	*	[53]	AP_VHV_EN
-	*	[54]	RCVR_CLK2
-	*	[55-61]	SDIO
-	*	[62]	NC
+	*	[17-26]	GPIO - Boot straps
+	*	[27-34]	GPIO
+	*	[35-36]	I2C1
+	*	[37-38]	I2C1
+	*	[39] 	GPIO
+	*	[40] 	RCVR CLK
+	*	[41]	GPIO VHV_EN
+	*	[43]	SD CARD DT
+	*	[44-55]	RGMII
+	*	[56-61]	SDIO
 	*/
 		   /*   0   1   2   3   4   5   6   7   8   9 */
-	pin-func = <	3   3   3   3   3   3   3   3   3   3
-			3   3   3   3   3   3   3   0   0   0
-			0   0   0   0   0   0   0   0   7   7
-			7   0   0   0   0   2   2   2   2   0
-			0   0   0   0   0   0   0   8   0   8
-			6   6   2   0   2   0xb  0xe 0xe 0xe 0xe
+	pin-func = <	0xa 0xa 8   8   6   6   9   9   9   0
+			0   0   3   3   3   3   3   0   0   0
+			0   0   0   0   0   0   0   0   0   0
+			0   0   0   0   0   2   2   2   2   0
+			2   0   0   1   1   1   1   1   1   1
+			1   1   1   1   1   1  0xe 0xe 0xe 0xe
 			0xe 0xe 0 >;
 };
 
@@ -156,7 +147,7 @@
 	clock-frequency = <100000>;
 	eeprom0: eeprom@50 {
                 compatible = "atmel,24c64";
-                reg = <0x50>;
+                reg = <0x53>;
                 pagesize = <0x20>;
         };
 	/*
@@ -195,11 +186,11 @@
 			input;
 			line-name = "pcie1.0-clkreq";
 		};
-		pcie1_0_w_disable {
+		pcie1_0_perst {
 			gpio-hog;
 			gpios = <3 GPIO_ACTIVE_LOW>;
 			output-low;
-			line-name = "pcie1.0-w-disable";
+			line-name = "pcie1.0-perst";
 		};
 		pcie2_0_clkreq {
 			gpio-hog;
@@ -207,12 +198,6 @@
 			input;
 			line-name = "pcie2.0-clkreq";
 		};
-		pcie2_0_w_disable {
-			gpio-hog;
-			gpios = <7 GPIO_ACTIVE_LOW>;
-			output-low;
-			line-name = "pcie2.0-w-disable";
-		};
 		usb3_ilimit {
 			gpio-hog;
 			gpios = <5 GPIO_ACTIVE_LOW>;
@@ -225,6 +210,20 @@
 			output-high;
 			line-name = "usb3-power";
 		};
+		pcie2_0_w_disable {
+			gpio-hog;
+			gpios = <8 GPIO_ACTIVE_LOW>;
+			output-low;
+			line-name = "pcie2.0-w-disable";
+		};
+
+		pcie2_0_gnss_disable {
+			gpio-hog;
+			gpios = <9 GPIO_ACTIVE_LOW>;
+			output-low;
+			line-name = "pcie2.0-gnss-disable";
+		};
+
 		m2_devslp {
 			gpio-hog;
 			gpios = <11 GPIO_ACTIVE_HIGH>;
@@ -359,10 +358,24 @@
 	};
 };
 
+&cp0_gpio1 {
+	phy_reset { /* Release switch reset */
+		gpio-hog;
+		gpios = <27 GPIO_ACTIVE_HIGH>;
+		output-high;
+	};
+};
+
 &cp0_ethernet {
 	status = "okay";
 };
 
+/* 1GE PHY over RGMII */
+&cp0_eth2 {
+	status = "okay";
+//	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+};
 
 &cp0_pinctl {
 	compatible = "marvell,mvebu-pinctrl", "marvell,armada-8k-cpm-pinctrl";
@@ -376,13 +389,18 @@
 		marvell,pins = < 35 36 >;
 		marvell,function = <2>;
 	};
+	cp0_ge2_rgmii_pins: cp0-ge-rgmii-pins-1 {
+                marvell,pins = < 44 45 46 47 48 49 50 51
+                                52 53 54 55 >;
+                marvell,function = <1>;
+	};
 	cp0_sdhci_pins: cp0-sdhi-pins-0 {
 		marvell,pins = < 56 57 58 59 60 61 >;
 		marvell,function = <14>;
 	};
 	cp0_sdhci_cd_pins: cp0-sdhci-cd-pins-0 {
-		marvell,pins = < 55 >;
-		marvell,function = <11>;
+		marvell,pins = < 43 >;
+		marvell,function = <1>;
 	};
 	cp0_spi1_pins: cp0-spi-pins-0 {
 		marvell,pins = < 13 14 15 16 >;
@@ -393,7 +411,7 @@
 		marvell,function = <3>;
 	};
 	cp0_mdio_pins: cp0-mdio-pins {
-                marvell,pins = < 47 49 >;
-                marvell,function = <8>;
+                marvell,pins = < 40 41 >;
+                marvell,function = <10>;
         };
 };
diff --git a/arch/arm/dts/cn9130-som.dtsi b/arch/arm/dts/cn9130-som.dtsi
index 51d56359aa..683821e0ea 100644
--- a/arch/arm/dts/cn9130-som.dtsi
+++ b/arch/arm/dts/cn9130-som.dtsi
@@ -30,6 +30,8 @@
 
 #include "armada-cp110.dtsi"
 
+#include "cn9130-db-dev-info.dtsi"
+
 / {
 	model = "SolidRun CN9130 based SOM";
 	compatible =	"marvell,armada70x0", "marvell,armada-ap806-quad",
-- 
2.25.1

